lib/rack/lock.rb

```diff
@@ -1,32 +1,27 @@
 # frozen_string_literal: true
 
-require 'thread'
+require 'logger'
 
 module Rack
-  # Rack::Lock locks every request inside a mutex, so that every request
-  # will effectively be executed synchronously.
+  # Sets up rack.logger to write to rack.errors stream
   class Lock
-    def initialize(app, mutex = Mutex.new)
-      @app, @mutex = app, mutex
+    def initialize(app)
+      @app = app
+      @mutex = ::Thread::Mutex.new
     end
 
     def call(env)
-      @mutex.lock
-      @env = env
-      @old_rack_multithread = env[RACK_MULTITHREAD]
-      begin
-        response = @app.call(env.merge!(RACK_MULTITHREAD => false))
-        returned = response << BodyProxy.new(response.pop) { unlock }
-      ensure
-        unlock unless returned
+      @mutex.synchronize do
+        @app.call(env)
       end
     end
 
-    private
-
-    def unlock
-      @mutex.unlock
-      @env[RACK_MULTITHREAD] = @old_rack_multithread
+    def self.rackup(config, app)
```

Comment:
I feel like this should be `app, config` simply because config is similar to `**options`. WDYT?

Reply:
`config` is first, because `rackup` handles args and block in addition to app. The code used in `Builder` is:

```ruby
        if middleware.respond_to?(:rackup)
          middleware.rackup(self, *args, &block) # self is the config in this case.
```

Reply:
Makes total sense, thanks for explaining.